docs/content/any/guides/data_filtering/index.md

```diff
@@ -0,0 +1,194 @@
+Value will be a list.
+- `Contains` means that the field must contain the value. This only applies
+if the field is a list.
+
+### Execute a Query
+
+`{{% exampleGet "execQuery" %}}` takes a query and returns a list of the results.
+
+### Combine Queries
+
+`{{% exampleGet "combineQuery" %}}` takes two queries and returns a new
+query that returns the union of the other two. For example if the two
+queries are SQL queries combine could `UNION` them. If they were HTTP
+requests `{{% exampleGet "combineQuery" %}}` could put them in an array and 
+`` could handle executing an array of queries
+and combining the results.
+
+You can define functions that apply to all types with
+`{{% exampleGet "setDataFilteringQueryDefaults" %}}`. Or you can pass type specific ones when you register a class.
+
+## Types
+
+The other thing you have to define to use data filtering are types for
+registered classes. These let oso know what the type of the fields are. Oso
+needs this information to handle specializers and other things in the policy
+when we don't have a concrete resource. The types are a 
+{{% exampleGet "map" %}} from field name to type.
+
+# Example
+{{< literalInclude
+      dynPath="exampleAPath"
+      from="docs: begin-a1"
+      to="docs: end-a1"
+      fallback="no" >}}
+
+For each class we need to register it and define the query functions.
+
+{{< literalInclude
+      dynPath="exampleAPath"
+      from="docs: begin-a2"
+      to="docs: end-a2"
+      fallback="no" >}}
+
+Then we can load a policy and query it.
+
+{{< literalInclude
+      dynPath="policyAPath"
+      fallback="no" >}}
+
+{{< literalInclude
+      dynPath="exampleAPath"
+      from="docs: begin-a3"
+      to="docs: end-a3"
+      fallback="no" >}}
+
+# Relations
+
+Often times you need data that is not contained on the object to make
+authorization decisions. This comes up when the role required to
+do something is implied by a role on it's parent object. For instance,
+you want to check the organization for a repository but that data isn't
+embedded on the repository object. You can add a `Relation` type to the type
+definition that states how the other resource is related to this one. Then
+you can access this field in the policy like any other field and it will
+fetch the data when it needs it (via the query functions).
+
+`Relation`s are a special type that tells oso how one Class is related to another. They specify what the related type is and how it's related.
+
+- `kind` is either "one" or "many". "one" means there is one related object and "many" means there is a list of related objects.
+- `other_type` is the class of the related objects.
+- `my_field` Is the field on this object that matches `other_field`.
+- `other_field` Is the field on the other object that matches `this_field`.
+
+The my_field / other_field relationship is similar to a foreign key. It lets oso know what fields to match up with building a query for the other type.
+
+# Example
+
+{{< literalInclude
```

Comment:
In SQLAlchemy users would usually setup sqlalchemy relationships between `Repository` & `Organization`.

Is there a reason you didn't do that?

Reply:
Sorta to show that we're not using that, we're using a Relation. It's kind of weird though.